lib/constrainer.js

```diff
@@ -38,19 +45,30 @@ class Constrainer {
       throw new Error(`There already exists a custom constraint with the name ${strategy.name}.`)
     }
 
-    if (this.strategiesInUse.has(strategy.name)) {
+    if (this.isStrategyUsed(strategy.name)) {
       throw new Error(`There already exists a route with ${strategy.name} constraint.`)
     }
 
     strategy.isCustom = true
+    strategy.isAsync = strategy.deriveConstraint.length === 3
```

Comment:
As developer, I would expect something like `strategy.deriveConstraint.constructor.name === 'AsyncFunction'` here

As user, I would expect the support for `async` functions

I'm PRO with a followup PR, but it should be more clear for the reader

Reply:
`strategy.deriveConstraint.length === 3` means when the function argument is 3. Then, it is using `callback` which is `async`.

I don't expect `async function` here since `async-await` is slow for a router.

Reply:
I know it, but new devs mix callback and async every time, and they expect an `async function` out of the box.

I learnt to avoid such confusion 😂 

> is slow for a router.

![image](https://user-images.githubusercontent.com/11404065/193640972-3dedf364-9d8f-44ee-add2-b7284d0119c4.png)

Reply:
> I know it, but new devs mix callback and async every time

Then, improve the document is better choice. And we should explain why we decide not to have `async function` here.

Reply:
@Eomm @climba03003 What did we decide? Are we good with the callback interface?

Reply:
> Are we good with the callback interface?

I believe yes, and we need a `caution` on the document about it doesn't support `async-await` and why we doesn't support it.